### PR TITLE
Console functional tests

### PR DESCRIPTION
**Pull request by ljmcgann**
Adds tests to node to ensure protection of console
related apis.

### CODE REVIEW

File: esiclient/tests/functional/v1/test_node.py [hunk 1]
```diff
@@ -137,3 +139,180 @@ def test_admin_can_node_power_on_off(self):
...
+        if node['provision_state'] != 'manageable':
+            utils.node_set_provision_state(self.clients['admin'],
+                                           self.node, 'manage')
+
+        utils.node_set(self.clients['admin'],
+                       self.node,
+                       'owner', self.projects['random']['id'])
+        self.addCleanup(utils.node_set,
+                        self.clients['admin'],
+                        self.node, 'owner')
```
Comment (tzumainn) [inline esiclient/tests/functional/v1/test_node.py:228]: You don't want to unset the owner; you want to set it to what it was previously by passing in node['owner']

File: esiclient/tests/functional/v1/test_node.py [hunk 2]
```diff
@@ -137,3 +139,180 @@ def test_admin_can_node_power_on_off(self):
...
+        if node['provision_state'] != 'manageable':
+            utils.node_set_provision_state(self.clients['admin'],
+                                           self.node, 'manage')
+
+        utils.node_set(self.clients['admin'],
+                       self.node,
+                       'lessee', self.projects['random']['id'])
+        self.addCleanup(utils.node_set,
+                        self.clients['admin'],
+                        self.node, 'lessee')
```
Comment (tzumainn) [inline esiclient/tests/functional/v1/test_node.py:274]: Same here; we should set this to node['lessee']

File: esiclient/tests/functional/v1/test_node.py [hunk 3]
```diff
@@ -137,3 +139,180 @@ def test_admin_can_node_power_on_off(self):
...
+        assert console['console_enabled']
+
+        utils.node_console_disable(self.clients['random-member'], self.node)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['random-member'],
+                                          self.node)
+        assert not console['console_enabled']
+
+    def test_random_cannot_enable_console(self):
+        """Tests lessee console functionality.
```
Comment (tzumainn) [inline esiclient/tests/functional/v1/test_node.py:293]: These comments may need to be updated

**Review by tzumainn — CHANGES_REQUESTED** [review]
Looks pretty good! A few comments inline.

File: esiclient/tests/functional/v1/test_node.py (hunk 3 above)
Comment (ljmcgann) [reply esiclient/tests/functional/v1/test_node.py:293]: fixed

File: esiclient/tests/functional/v1/test_node.py (hunk 2 above)
Comment (ljmcgann) [reply esiclient/tests/functional/v1/test_node.py:274]: fixed

File: esiclient/tests/functional/v1/test_node.py (hunk 1 above)
Comment (ljmcgann) [reply esiclient/tests/functional/v1/test_node.py:228]: fixed

**Comment by ljmcgann** [comment]
Thinking about it, i think the sleeps might not actually be needed. We are only testing the protection of apis and not whether the console actually works. So it should be sufficient to just run the enable, disable, and show commands, and if they dont return an error, then we call it a success

File: esiclient/tests/functional/README.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ These tests are designed to ensure Ironic and Neutron behave as expected when co
 ### Prerequisites
 
 These tests are intended to be ran against a functioning OpenStack cloud with Ironic and Neutron services enabled and running. It assumes that the ironic.conf file
-is located at `/etc/ironic/ironic.conf`. These tests assume that the `fake-hardware` hardware type is enabled under the `enabled_hardware_types` option under `[DEFAULT]` heading.
+is located at `/etc/ironic/ironic.conf`. These tests assume that the `fake-hardware` hardware type is enabled under the `enabled_hardware_types` option under `[DEFAULT]` heading. Some of these tests require an Ironic node configured with ipmi credentials in order to test the use of the ipmitool-socat serial console.
```
Comment (tzumainn) [inline esiclient/tests/functional/README.md:8]: May as well link to the Ironic documentation page describing what's needed.

File: esiclient/tests/functional/v1/test_node.py (hunk 3 above)
Comment (tzumainn) [reply esiclient/tests/functional/v1/test_node.py:293]: Ah, I think you may have forgotten to update this test description!

**Review by tzumainn — COMMENTED** [review]
Tests pass for me! I think the last two document-ish comments are the only other things that I'd like changed.

File: esiclient/tests/functional/README.md (hunk 4 above)
Comment (ljmcgann) [reply esiclient/tests/functional/README.md:8]: added

**Review by tzumainn — APPROVED** [review]
Looks good!

**Pull request merged**